webpack/assets/javascripts/react_app/components/ForemanModal/subcomponents/SubmitOrCancel/SubmitOrCancelActions.js

```diff
@@ -0,0 +1,56 @@
+import { API } from '../../../../redux/API';
+
+import { addToast } from '../../../../redux/actions/toasts';
+import { translate as __ } from '../../../../common/I18n';
+import {
+  setModalStartSubmitting,
+  setModalStopSubmitting,
+} from '../../ForemanModalActions';
+
+const onModalError = error => {
+  const { response } = error;
+
+  if (response.data && response.data.error) {
+    if (response.data.error.message) {
+      return response.data.error.message;
+    }
+
+    if (response.data.error.full_messages) {
+      return response.data.error.full_messages.join(', ');
+    }
+  }
+
+  return `${error.response.status}: ${__('Failed to submit the request.')}`;
```

Comment:
I believe you can destruct the error object into something like: 
`const { response: { status, data: { error: { message, full_messages } } } = {} } = error`

and it will make the code below look much cleaner.